File: Google_Kick_Start/2020/Round_A/B.cpp
#include <bits/stdc++.h>
using namespace std;
using namespace std;
typedef std::stack<int> Stack;
typedef std::vector<int> Array;

void find_max (Stack stacks[], int N, Array &arr, int M) {
arr.resize(M+1, 0);
for (int i = 0; i < N; ++i) {
int index = 0;
int sum = 0;
while (!stacks[i].empty()) {
sum += stacks[i].top(); ++index;
if (arr[index] < sum) arr[index] = sum;
stacks[i].pop();
}
} //end of all stacks iteration
}

int find_max_sum (Array &arr, int M) {
int max_sum = 0;
for (int ind = 0; ind < M/2; ++ind) {
int first = ind;
int last = M - first;
if (arr[first] + arr[last] > max_sum) max_sum = arr[first] + arr[last];
}
return max_sum;
}

int main(){
    ios_base::sync_with_stdio(0);
    cin.tie(NULL); cout.tie(NULL);
    int T;
    cin>>T;
    for(int test =1; test<=T;test++){
        int n,k,p;
        cin>>n>>k>>p;
        vector <vector<int>> stacks(n, vector<int>(k));
        for(int i=0;i<n;i++)
            for(int j=k-1;j>=0;j--)
                cin>>stacks[i][j];
        cout<<"Case #"<<test<<": "<<maxSum(stacks,p,n,k)<<"\n";
    }
    return 0;
}

File: Google_Kick_Start/2020/Round_A/B.py
def find_max_sum(s_arr, n):
    if n == 0:
        return 0
    
    if len(s_arr) == 1:
        return sum(s_arr[0][-n:])
    max_sum = find_max_sum(s_arr[1:], n)
    
    first_stack = s_arr[0]
    for i in range(1,n+1): 
        stack_sum = sum(first_stack[-i:])
        rem_sum = find_max_sum(s_arr[1:], n-i)
        if stack_sum + rem_sum > max_sum:
            max_sum = stack_sum + rem_sum
    
    return max_sum

def knapSack(arr,W):
    n = len(arr)
    k = len(arr[0])
    dp = [[0 for _ in range(W+1)] for _ in range(n+1)]
    for i in range(n+1):
        for j in range(W+1):
            if i==0 or j==0:
                continue
            else:
                pass
    
def main():
    t = int(input())
    for test in range(1,t+1):
        n,k,p = map(int,input().split())
        value = [0]
        wt = [0]
        for _ in range(n):
            tmp = list(map(int,input().split()))
            value.append(tmp[0])
            wt.append(1)
            for j in range(1,k):
                tmp[j] += tmp[j-1]
                value.append(tmp[j])
                wt.append(j+1)
        print(value,wt)
        print("Case #{}: {}".format(test,knapSack(value,wt)))
if __name__ == "__main__":
    main()